src/PluginsSystem/Tools/Microsoft.Performance.Toolkit.Plugins.Cli/Metadata/MetadataGenerator.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Data;
+using Microsoft.Extensions.Logging;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Manifest;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Validation;
+using Microsoft.Performance.Toolkit.Plugins.Core;
+using Microsoft.Performance.Toolkit.Plugins.Core.Metadata;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Cli.Metadata
+{
+    public sealed class MetadataGenerator
+        : IMetadataGenerator
+    {
+        private readonly ILogger<MetadataGenerator> logger;
+
+        public MetadataGenerator(ILogger<MetadataGenerator> logger)
+        {
+            this.logger = logger;
+        }
+
+        public PluginMetadata Generate(ProcessedPluginDirectory pluginDirectory, PluginManifest manifest)
+        {
+            PluginIdentity identity = new(manifest.Identity.Id, manifest.Identity.Version);
```

Comment:
We should probably augment a lot of these methods with logger.Verbose calls so we can track what happens